mantidproject/mantid
# Reflectometry GUI restructuring: Plotting

**Description of work.**
The new restructure of the GUI does not have plotting functionality. As this is a major part of the Porting of the GUI to Workbench, I have added it ahead of porting it to workbench.

**To test:**

**Note for testsing, the data will look different but that is not the fault of this PR**

### First, set up some processed rows
- Open the `ISIS Reflectometry` interface, select `INTER`.
- From the Reflectometry menu, select Import table and choose `INTER_NR_test2.tbl` from the ISIS sample data.
- Click `Process` and when finished expand all groups. All rows should now be green.

### Test the first button ("plot runs")
- Select a single row within a group and then click plot runs. You get a single curve:
<img src="https://user-images.githubusercontent.com/12895056/55162331-82d12b80-515f-11e9-8dd0-3a49809f1bf9.png" width=200>

- Select two rows and click plot runs. You get two curves on one plot:
<img src="https://user-images.githubusercontent.com/12895056/55162520-f115ee00-515f-11e9-8dec-858213b44066.png" width=200>

- Select a group and click plot runs. You get all of the curves from the group on one plot:
<img src="https://user-images.githubusercontent.com/12895056/55162568-068b1800-5160-11e9-81f1-d1a1915ef2df.png" width=200>

- Select one row from the first group and two rows from the second group. You should get three curves:
<img src="https://user-images.githubusercontent.com/12895056/55162805-7ef1d900-5160-11e9-8a78-c1985b377c20.png" width=200>

- Select one row from the first group and the overall group for the second group. You should get three curves:
<img src="https://user-images.githubusercontent.com/12895056/55162881-a779d300-5160-11e9-8977-0c863f116e61.png" width=200>


### Test the second button ("plot group")
- Select a row or rows within a group and click plot group. Nothing should happen (ideally this should be disabled).

- Select a group and click plot group. You should get a single curve (for the stitched result):
<img src="https://user-images.githubusercontent.com/12895056/55162725-52d65800-5160-11e9-9c48-ebdda990ad04.png" width=200>

- Select two groups and click plot group. You should get two curves:
<img src="https://user-images.githubusercontent.com/12895056/55162761-65509180-5160-11e9-96d4-9085292fd215.png" width=200>

- Select one row from the first group and the overall group for the second group. The row should be ignored and one curve plotted for the second group:
<img src="https://user-images.githubusercontent.com/12895056/55162950-c7a99200-5160-11e9-8481-6873c3831ffc.png" width=200>

*Tests posted by @gemmaguest on the issue*



Fixes #25431 . 


*This does not need release notes as I believe it is an expected feature of the new GUI already*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
